job_creator.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+class JobCreator
+{
+    public function getInstallerVersion($githubRepository, $branch)
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        $v = explode('.', $branch);
+        if (count($v) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
```

Comment:
I don't this so, let's say we're doing a pull-request for asset-admin targetting the `1` branch.  We want the `4` of installer, so the count makes sense

If we're targetting `1.10`, then we want installer `4.10`

Reply:
It ultimately doesn't matter given that you pointed out we can't take the major number from the repo's major version as it may not match.

I still think the count should be a fallback, but it doesn't make an actual difference in any situation I can think of so it's probably okay to leave as is.